File: doc/filter.lua
local text = require('text')

function Header(el)
  -- drop level-1 header
  if el.level == 1 then
    return {}
  else
    -- decrease level of all headers by 1
    el.level = el.level - 1
    -- convert level-1 headers to uppercase
    if el.level == 1 then
      return pandoc.walk_block(el, {
        Str = function(el)
          return pandoc.Str(text.upper(el.text))
      end })
    end
  end
end

File: doc/filter.py
#!/usr/bin/env python3

import fileinput
import os
import sys

for line in fileinput.input():
    # substitute package version
    if line.startswith("% "):
        line = line.replace("@PACKAGE_VERSION@", os.environ["PACKAGE_VERSION"])
    # drop level-1 header
    if line.startswith("# "):
        continue
    # decrease level of all headers by 1
    if line.startswith("##"):
        line = line.replace("#", "", 1)
    # convert level-1 headers to uppercase
    if line.startswith("# "):
        line = line.upper()
    sys.stdout.write(line)
